influxql/engine.go

```diff
@@ -744,6 +746,54 @@ func ReduceMax(key Key, values []interface{}, e *Emitter) {
 	}
 }
 
+// MapStddev collects the values to pass to the reducer
+func MapStddev(itr Iterator, e *Emitter, tmax int64) {
+	var values []float64
+
+	for k, v := itr.Next(); k != 0; k, v = itr.Next() {
```

Comment:
You're already putting all the values into a single emit. I would just say that you need to limit the size of that. So you can do multiple emits per map